packages/ripple-binary-codec/src/types/st-array.ts

```diff
@@ -14,7 +14,13 @@ const OBJECT_END_MARKER = Uint8Array.from([0xe1])
  */
 function isObjects(args): args is Array<JsonObject> {
   return (
-    Array.isArray(args) && (args.length === 0 || typeof args[0] === 'object')
+    Array.isArray(args) &&
+    args.every(
+      (arg) =>
+        typeof arg === 'object' &&
+        Object.keys(arg).length === 1 &&
```

Comment:
could you explain why we need lines 21 and 22?

```
{
"key1": "value1",
"key2": "value2"
}
```

An input like this is rejected because there are 2 keys, and the values are "string", not "object". Why do we need this condition?

Reply:
That's not what valid inner objects look like.
This is what they must look like:
```
{
  "PriceData": {
    "BaseAsset": "XRP",
    "QuoteAsset": "USD",
    "AssetPrice": 740,
    "Scale": 3
  }
}
```

Reply:
hmm okay. this format slipped my mind 👍

Reply:
thanks